---
 src/ui/theme-parser.c |   38 ++++++++++++++++++++++++++++++++++++--
 1 file changed, 36 insertions(+), 2 deletions(-)

--- a/src/ui/theme-parser.c
+++ b/src/ui/theme-parser.c
@@ -88,7 +88,9 @@ typedef enum
   STATE_WINDOW,
   /* things we don't use any more but we can still parse: */
   STATE_MENU_ICON,
-  STATE_FALLBACK
+  STATE_FALLBACK,
+  /* an ubuntu specific ignore-this-element state */
+  UBUNTU_STATE_IGNORE  
 } ParseState;
 
 typedef struct
@@ -1308,7 +1310,19 @@ parse_toplevel_element (GMarkupParseCont
        */
       push_state (info, STATE_FALLBACK);
     }
-   else
+  else if (ELEMENT_IS ("shadow"))
+    {
+       /* ubuntu specific, workaround for light-themes: silently ignore shadow tag.
+        */
+      push_state (info, UBUNTU_STATE_IGNORE);
+    }
+  else if (ELEMENT_IS ("padding"))
+    {
+       /* ubuntu specific, workaround for light-themes: silently ignore padding tag.
+        */
+      push_state (info, UBUNTU_STATE_IGNORE);
+    }
+  else
     {
       set_error (error, context,
                  G_MARKUP_ERROR, G_MARKUP_ERROR_PARSE,
@@ -3029,6 +3043,18 @@ parse_style_element (GMarkupParseContext
       
       push_state (info, STATE_BUTTON);
     }
+  else if (ELEMENT_IS ("shadow"))
+    {
+       /* ubuntu specific, workaround for light-themes: silently ignore shadow tag.
+        */
+      push_state (info, UBUNTU_STATE_IGNORE);
+    }
+  else if (ELEMENT_IS ("padding"))
+    {
+       /* ubuntu specific, workaround for light-themes: silently ignore padding tag.
+        */
+      push_state (info, UBUNTU_STATE_IGNORE);
+    }
   else
     {
       set_error (error, context,
@@ -3670,6 +3696,8 @@ start_element_handler (GMarkupParseConte
                  _("Element <%s> is not allowed inside a <%s> element"),
                  element_name, "fallback");
       break;
+    case UBUNTU_STATE_IGNORE:
+      break;
     }
 }
 
@@ -3959,6 +3987,9 @@ end_element_handler (GMarkupParseContext
       pop_state (info);
       g_assert (peek_state (info) == STATE_THEME);
       break;
+    case UBUNTU_STATE_IGNORE:
+      pop_state (info);
+      break;
     }
 
   pop_required_version (info);
@@ -4164,6 +4195,9 @@ text_handler (GMarkupParseContext *conte
     case STATE_FALLBACK:
       NO_TEXT ("fallback");
       break;
+    case UBUNTU_STATE_IGNORE:
+      NO_TEXT ("ignored_element");
+      break;
     }
 }
 
