From e3d5969282e4fad90161287781eb11f6c83d0d67 Mon Sep 17 00:00:00 2001
From: "Jasper St. Pierre" <jstpierre@mecheye.net>
Date: Thu, 7 Aug 2014 13:39:21 -0400
Subject: [PATCH 1/7] background: Don't bother supporting alpha channels for
 gradients

---
 src/compositor/meta-background.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/compositor/meta-background.c b/src/compositor/meta-background.c
index d143b21..0cd2eae 100644
--- a/src/compositor/meta-background.c
+++ b/src/compositor/meta-background.c
@@ -814,15 +814,15 @@ meta_background_load_gradient (MetaBackground             *self,
   pixels[0] = color->red;
   pixels[1] = color->green;
   pixels[2] = color->blue;
-  pixels[3] = color->alpha;
+  pixels[3] = 0xFF;
   pixels[4] = second_color->red;
   pixels[5] = second_color->green;
   pixels[6] = second_color->blue;
-  pixels[7] = second_color->alpha;
+  pixels[7] = 0xFF;
 
   texture = cogl_texture_new_from_data (width, height,
                                         COGL_TEXTURE_NO_SLICING,
-                                        COGL_PIXEL_FORMAT_RGBA_8888,
+                                        COGL_PIXEL_FORMAT_RGB_888,
                                         COGL_PIXEL_FORMAT_ANY,
                                         4,
                                         pixels);
-- 
2.1.0.rc1

